modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/size_items.dm

```diff
@@ -0,0 +1,129 @@
+	. = ..()
+	if(!warning_given)
+		if(tgui_alert(user, "This item is strictly intended as an ERP item for use in dorm rooms. Failure to respect this will result in administrative action being taken. Do you wish to continue using this item?", "A word of warning.", list("Yes", "No")) != "Yes")
+			return FALSE
+
+		warning_given = TRUE
+
+	var/chosen_size = tgui_input_number(user, "What size percentage do you wish to set the collar to?", name, 100, CONFIG_GET(number/size_collar_maximum), CONFIG_GET(number/size_collar_minimum))
+	if(!chosen_size)
+		balloon_alert(user, "invalid size!")
+		return FALSE
+
+	log_message("[src] had its target size changed to [chosen_size]% by [usr]", LOG_ATTACK)
+	balloon_alert(user, "set to [chosen_size]%")
+	target_size = (chosen_size * 0.01)
+	return TRUE
+
+/obj/item/clothing/neck/size_collar/mob_can_equip(mob/living/user, slot, disable_warning, bypass_equip_delay_self, ignore_equipped)
+	if(!warning_given)
+		return FALSE
+
+	return ..()
+
+/obj/item/clothing/neck/size_collar/equipped(mob/living/user, slot)
+	. = ..()
+	if(!ishuman(user) || !(slot & ITEM_SLOT_NECK))
+		return FALSE
+
+	size_component = user.AddComponent(/datum/component/temporary_size, target_size)
+	size_component.target_size = target_size
+
+	user.log_message("[src] was equipped by [user].", LOG_ATTACK)
+
+/obj/item/clothing/neck/size_collar/dropped(mob/living/user)
+	. = ..()
+	if(size_component)
+		qdel(size_component)
+		size_component = null
+
+/obj/item/clothing/neck/size_collar/examine(mob/user)
+	. = ..()
+	var/list/area_names = list()
+	for(var/area_index in SIZE_WHITELISTED_AREAS) //We can't do this typed.
+		var/area/area_type = area_index //So we have to assign it to a typed variable after we get it from the loop.
+		var/area_name = initial(area_type.name)
+		if(!area_name)
+			continue
+
+		area_names += area_name
+
+	if(length(area_names))
+		. += span_cyan("This collar will work in the following areas: [english_list(area_names)]")
+
+	return .
+
+/// Component that temporarily applies a size to a human.
+/datum/component/temporary_size
+	/// List containing the areas that the size change works in. If this is empty, this will work everywhere.
+	var/list/allowed_areas = SIZE_WHITELISTED_AREAS
+	/// What is the stored size of the mob using this?
+	var/original_size = 1
+	/// What size are we changing the parent mob to?
+	var/target_size = 1
+
+/datum/component/temporary_size/Initialize(size_to_apply)
+	. = ..()
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	var/mob/living/carbon/human/human_parent = parent
+	original_size = human_parent?.dna.features["body_size"]
+
+	if(!original_size) //If we aren't able to get the original size, we shouldn't exist.
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_ENTER_AREA, .proc/check_area)
+
+	target_size = size_to_apply
+	check_area()
+
+/// Checks if we need to revert our size when entering a different area.
+/datum/component/temporary_size/proc/check_area()
```

Comment:
Does this mean that it would revert back to the target size as soon as you step back into the area, though? Just making sure I understand the goal of that bit in particular.

Reply:
Correct. So this is called whenever someone enters an area, and if the area is compatible, they are turned into the target size on the collar, otherwise they are turned to their normal size.

Reply:
Hmmm. I know I'm not going to be using that feature, but I feel like maybe it would be a good idea that, if it turns off when they go out, that they'd just need to re-enable it when they get back in? Otherwise I could see it getting a little awkward if someone somehow forgot to turn it off or whatever before getting back in, after having a scene earlier into the shift, for instance. What do you think?